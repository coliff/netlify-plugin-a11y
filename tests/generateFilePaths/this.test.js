const fs = require('fs')
const path = require('path')

const PUBLISH_DIR = path.join(__dirname, 'publishDir')
// actual test
const pluginCore = require('../../plugin/pluginCore.js')
test('generateFilePaths works', async () => {
	const results = await pluginCore.generateFilePaths({
		absolutePublishDir: PUBLISH_DIR,
		fileAndDirPaths: ['/blog', '/about.html'],
		ignoreDirectories: [],
	})
	expect(results).toMatchSnapshot()
})

const pathInResults = (expectedPath, results) => {
	return results.findIndex((r) => r.endsWith(expectedPath)) != -1
}

test('ignoreDirectories works including leading slash', async () => {
	const results = await pluginCore.generateFilePaths({
		fileAndDirPaths: ['/'],
		ignoreDirectories: ['/admin'],
		absolutePublishDir: PUBLISH_DIR,
	})
	expect(pathInResults('publishDir/blog/post1.html', results)).toBe(true)
	expect(pathInResults('publishDir/about.html', results)).toBe(true)
	expect(pathInResults('publishDir/index.html', results)).toBe(true)
	expect(pathInResults('publishDir/admin/index.html', results)).toBe(false)
})

test('ignoreDirectories works without leading slash', async () => {
	const results = await pluginCore.generateFilePaths({
		fileAndDirPaths: ['/'],
		ignoreDirectories: ['admin'],
		absolutePublishDir: PUBLISH_DIR,
	})
	expect(pathInResults('publishDir/blog/post1.html', results)).toBe(true)
	expect(pathInResults('publishDir/about.html', results)).toBe(true)
	expect(pathInResults('publishDir/index.html', results)).toBe(true)
	expect(pathInResults('publishDir/admin/index.html', results)).toBe(false)
})
