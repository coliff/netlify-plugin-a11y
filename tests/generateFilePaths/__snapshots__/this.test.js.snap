// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateFilePaths works 1`] = `
Array [
  "localhost:9000/tests/generateFilePaths/publishDir/blog/post1.html",
  "localhost:9000/tests/generateFilePaths/publishDir/blog/post2.html",
  "localhost:9000/tests/generateFilePaths/publishDir/about.html",
]
`;
